test/test_tree.cpp

```diff
@@ -987,6 +1003,93 @@ TEST(Tree, operator_square_brackets)
     }
 }
 
+TEST(Tree, noderef_at)
+{
+    {
+        Tree t = parse_in_arena("[0, 1, 2, 3, 4]");
+        NodeRef m = t.rootref();
+        ConstNodeRef const cm = t.rootref();
+        EXPECT_EQ(m.at(0).val(), "0");
+        EXPECT_EQ(m.at(1).val(), "1");
+        EXPECT_EQ(m.at(2).val(), "2");
+        EXPECT_EQ(m.at(3).val(), "3");
+        EXPECT_EQ(m.at(4).val(), "4");
+        EXPECT_EQ(cm.at(0).val(), "0");
+        EXPECT_EQ(cm.at(1).val(), "1");
+        EXPECT_EQ(cm.at(2).val(), "2");
+        EXPECT_EQ(cm.at(3).val(), "3");
+        EXPECT_EQ(cm.at(4).val(), "4");
+        //
+        EXPECT_TRUE(m.at(0)  == "0");
+        EXPECT_TRUE(m.at(1)  == "1");
+        EXPECT_TRUE(m.at(2)  == "2");
+        EXPECT_TRUE(m.at(3)  == "3");
+        EXPECT_TRUE(m.at(4)  == "4");
+        EXPECT_TRUE(cm.at(0) == "0");
+        EXPECT_TRUE(cm.at(1) == "1");
+        EXPECT_TRUE(cm.at(2) == "2");
+        EXPECT_TRUE(cm.at(3) == "3");
+        EXPECT_TRUE(cm.at(4) == "4");
+        //
+        EXPECT_FALSE(m.at(0)  != "0");
+        EXPECT_FALSE(m.at(1)  != "1");
+        EXPECT_FALSE(m.at(2)  != "2");
+        EXPECT_FALSE(m.at(3)  != "3");
+        EXPECT_FALSE(m.at(4)  != "4");
+        EXPECT_FALSE(cm.at(0) != "0");
+        EXPECT_FALSE(cm.at(1) != "1");
+        EXPECT_FALSE(cm.at(2) != "2");
+        EXPECT_FALSE(cm.at(3) != "3");
+        EXPECT_FALSE(cm.at(4) != "4");
+        //
+        //TODO: Not sure what to replace with capacity
+        //verify_error(t, [&](Tree const&){ return cm[m.capacity()]; });
```

Comment:
Asking for a node==capacity (which doesn't exist) should fail. Leave it here, I will fix it after merging.

Reply:
Cool